repos/system_upgrade/el7toel8/actors/sourcebootloaderscanner/libraries/sourcebootloaderscanner.py

```diff
@@ -26,20 +50,28 @@ def scan_boot_entries():
         )
 
     boot_entries = []
-    # Identify the available boot entries by searching for their titles in the grubby output
+    boot_entry_data = None
     for output_line in grubby_output['stdout']:
-        # For now it is sufficient to look only for the titles as that is the only
-        # information we use. If need be, we would have to parse the structure
-        # of the grubby output into sections according to the `index` lines
-        if output_line.startswith('title='):
-            boot_entry = output_line[6:]  # Remove the `title=` prefix
-
-            # On s390x grubby produces quotes only when needed (whitespace in
-            # the configuration values), on x86 the values are quoted either way
-            boot_entry = boot_entry.strip('\'"')
+        if output_line == 'non linux entry':
+            # Grubby does not display info about non-linux entries
+            # Such an entry is not problematic from our PoV, therefore, skip it
+            boot_entry_data = {}
+            continue
 
-            boot_entries.append(BootEntry(title=boot_entry))
+        prop, value = parse_grubby_output_line(output_line)
+        if prop == 'index':
+            # Start of a new boot entry section
+            if boot_entry_data is not None:
```

Comment:
Nit: you can have `if boot_entry_data:` here

Reply:
I remember that there was a reason why I needed to explicitly want `None`, however I do not see it now. I've fixed it.

This also fixes the possible `KeyError` as an empty dictionary resulting from a non-linux entry will be treated as `False`, and therefore, this branch will not be taken.